examples/uefi-demo/main.rs

```diff
@@ -133,8 +165,15 @@ fn wait_for_input(max_timeout: Option<Duration>) {
     {
         // SAFETY: The cloned handles are only used to wait for further input events and
         // are then immediately dropped.
-        let mut events =
-            unsafe { [st().stdin().wait_for_key_event().unsafe_clone(), timer.unsafe_clone()] };
+        let ptr = POINTER.load(Ordering::Relaxed);
+        let pointer_ref = unsafe { &*ptr };
```

Comment:
we should remove this unsafe here if we store the Box instead of a raw pointer.

Reply:
> we should remove this unsafe here if we store the Box instead of a raw pointer.
> why do we make it a raw pointer instead of just storing the Box?

help me to fix it please

Reply:
This just doesn't look safe.

Could you perhaps call get_handle_for_protocol and get_handle_for_protocol everytime, or are these expensive calls?